src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/cost/DefaultTripSchedule.java

```diff
@@ -18,7 +18,13 @@ public interface DefaultTripSchedule extends RaptorTripSchedule {
 
   /**
    * This is not used by the default calculator, but by the {@link RouteCostCalculator} to
-   * give unpreferred routes or agencies a generalized-cost penalty.
+   * give unpreferred routes a generalized-cost penalty.
    */
   FeedScopedId routeId();
+
+  /**
+   * This is not used by the default calculator, but by the {@link AgencyCostCalculator} to
+   * give unpreferred agencies a generalized-cost penalty.
+   */
+  FeedScopedId agencyId();
```

Comment:
I am not sure you need this, this will slow down the search in general - even without the filter set. You should be able to filter out the unpreferred routes based on the agency. This will make the search with the unpreferred agency slower, but not other searches. I would test these two ways of solving the problem to look at the performance overhead.